### PR TITLE
feat(inputs.snmp): Convert enum values

### LINKED ISSUE #2809: [feature-request] support SNMP enum values
### Proposal:

Is there any way to get Telegraf SNMP plugin to collect the enum string value rather than the integer type?

### Current behavior:

For example, I'm collecting from IF-MIB::ifType and it has a bunch of values defined in an enum table. I would rather store that information in InfluxDB rather than the integer type.

### Desired behavior:

This could be added to the 'conversion' option, but basically instead of returning an integer for the enum, return the string

### Use case: [Why is this important (helps with prioritizing requests)]

The enum integer value is meaningless for those unfamilar with SNMP. Most users rather have the string value that the integer represents.

### LINKED ISSUE #6330: SNMP integer fields are not translated
It seems that integer fields that have translations available, are actually not translated. I thought it should have been set in the field `conversion` parameter, but it seems there is no `translate` or similar option.

It seems like I could work around by using something as the reverse of the enum processor plugin. (not sure if this is possible)

### Relevant telegraf.conf:
```toml
  [[inputs.snmp.table]]
    name = "interface"

    [[inputs.snmp.table.field]]
      oid = "IF-MIB::ifName"
      is_tag = true

    [[inputs.snmp.table.field]]
      oid = "IF-MIB::ifAdminStatus"
      is_tag = true

    [[inputs.snmp.table.field]]
      name = "ifInOctets"
      oid = "IF-MIB::ifHCInOctets"

    [[inputs.snmp.table.field]]
      name = "ifOutOctets"
      oid = "IF-MIB::ifHCOutOctets"
```

### System info:

Telegraf 1.11.4 (git: HEAD d9ca76e4)
RedHat 7.4

### Expected behavior:
```
> interface,ifAdminStatus=up(1),ifName=InLoopBack0 ifInOctets=0i,ifOutOctets=0i
```
### Actual behavior:
```
> interface,ifAdminStatus=1,ifName=InLoopBack0 ifInOctets=0i,ifOutOctets=0i
```
### Additional info:
Mib:
```mib
ifAdminStatus OBJECT-TYPE
    SYNTAX INTEGER {
        up(1),
        down(2),
        testing(3)
        }
    ACCESS read-write
    STATUS mandatory
    DESCRIPTION
        "The desired state of the interface.  The testing(3) state
        indicates that no operational packets can be passed.  When a
        managed system initializes, all interfaces start with
        ifAdminStatus in the down(2) state.  As a result of either
        explicit management action or per configuration information
        retained by the managed system, ifAdminStatus is then
        changed to either the up(1) or testing(3) states (or remains
        in the down(2) state)."
    ::= { ifEntry 7 }
```
Output from `snmpget`:
```
IF-MIB::ifAdminStatus.1 = INTEGER: up(1)
```

### PR DESCRIPTION
**Pull request by Hipska**
# Required for all PRs

- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #6330
resolves #2809

Adds new options `enum` or `enum(1)` for the field `conversion` parameter.

### CODE REVIEW

**Review by MyaLongmire — COMMENTED** [review]
@Hipska the direction of this is great! my only comment so far is to document the conversion options get documented in the readme when you get there :)

**Comment by LeadingMoominExpert** [comment]
@Hipska good stuff 👏🏻

File: plugins/inputs/snmp/snmp.go [hunk 1]
```diff
@@ -752,6 +760,14 @@ func fieldConvert(conv string, v interface{}) (interface{}, error) {
 		return v, nil
 	}
 
+	if conv == "enum" {
+		return tr.SnmpFormatEnum(ent.Name, ent.Value, false)
+	}
+
+	if conv == "enum(1)" {
+		return tr.SnmpFormatEnum(ent.Name, ent.Value, true)
+	}
```
Comment (srebhan) [inline plugins/inputs/snmp/snmp.go:769]: This whole thing should really be a `switch` statement, but I don't want you to change it in this PR. So please ignore this comment for this PR.

**Review by srebhan — COMMENTED** [review]
This looks very nice @Hipska! Just have one request on changing the returned error to a `nil` explicitly to help the reader to easier catch the non-error nature...

File: plugins/inputs/snmp/gosmi.go [hunk 2]
```diff
@@ -122,3 +123,21 @@ func (g *gosmiTranslator) SnmpTableCall(oid string) (mibName string, oidNum stri
...
+	}
+
+	var v models.Value
+	if full {
+		v = node.FormatValue(value, models.FormatEnumName, models.FormatEnumValue)
+	} else {
+		v = node.FormatValue(value, models.FormatEnumName)
+	}
+
+	return v.Formatted, err
```
[the remark this answers is not in the document]
Comment (Hipska) [reply plugins/inputs/snmp/gosmi.go:142]: Sure, saw it as well yesterday an thought you would say something about it 😅

File: plugins/inputs/snmp/snmp.go (hunk 1 above)
Comment (Hipska) [reply plugins/inputs/snmp/snmp.go:769]: I can do it if you want, but what about that one case which is doing `scanf`?

Comment (srebhan) [reply plugins/inputs/snmp/snmp.go:769]: Don't do it here. But we really should think about converting this in another PR. The exceptions can be handled in default or in if cases after/before the switch statement...

**Review by powersj — APPROVED** [review]

**Pull request merged**